src/Interpreters/Session.h

```diff
@@ -49,6 +50,7 @@ class Session
     /// Sets the current user, checks the credentials and that the specified address is allowed to connect from.
     /// The function throws an exception if there is no such user or password is wrong.
     void authenticate(const String & user_name, const String & password, const Poco::Net::SocketAddress & address);
+    void authenticate(const String & user_name, const String & password, const Poco::Net::StreamSocket & socket);
```

Comment:
No, we don't need this overload, `HTTPHandler` should pass the current certificate's common name via passing `CertificateCredentials` as `credentials`, and not via `socket`.